mne/tests/test_filter.py

```diff
@@ -1107,3 +1105,32 @@ def test_filter_minimum_phase_bug():
     dB_min_half = 20 * np.log10(np.abs(H_min_half[mask]))
     assert_array_less(dB_min_half, -20)
     assert not (dB_min_half < -30).all()
+
+
+@pytest.mark.parametrize("dc", (0, 100))
+@pytest.mark.parametrize("sfreq", (1000.0, 999.0))
+def test_smart_pad(dc, sfreq):
+    """Test that smart pad does what it should."""
+    f = 1.0
+    t = (np.arange(0, int(round(sfreq)) + 1)) / sfreq
+    x = np.sin(2 * np.pi * f * t) + dc
+    x_want = np.r_[x, x, x]
+    padlen = (len(x),) * 2
+    x_pad = np.pad(x, padlen, mode="reflect", reflect_type="odd")
+    assert_allclose(x_pad, x_want, err_msg="np.pad", atol=0.1)  # slight DC shift
+    this_x_pad = _smart_pad(x, padlen, "reflect")
+    # slight DC shift from out "x_want" (doubled sample at each end)
+    assert_allclose(this_x_pad, x_want, atol=0.1, err_msg="_smart_pad reflect x_want")
+    assert_allclose(
+        this_x_pad, x_pad, atol=1e-6, err_msg="_smart_pad reflect vs np.pad"
+    )
+    this_x_pad = _smart_pad(x, padlen, "reflect_limited")
+    x_want = np.r_[0, x[:-1], x, x[1:], 0]
+    assert_allclose(
+        this_x_pad, x_want, atol=1e-7, err_msg="_smart_pad reflect_limited x_want"
+    )
+    # reflect_limited uses one fewer sample so ends up a little bit different
+    # with even more padding
+    x_want = np.r_[np.full(len(x), 0), x_want, np.full(len(x), 0)]
```

Comment:
is there a reason here to prefer `np.full(len(x), 0)` instead of `np.zeros(len(x))` or `np.zeros_like(x)`?